lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/PostgresKgQueryStore.scala

```diff
@@ -35,42 +70,78 @@ final class PostgresKgQueryStore @Inject()(configProvider: PostgresStoreConfigPr
       .headOption
   }
 
-  override def getNodeContext(id: String): Option[KgNodeContext] = None
+  override def getNodeContext(id: String): Option[KgNodeContext] =
+    runSyncTransaction(nodes.getById(id)).map { _ =>
+      val nodeLabelQuery = for {
+        nodeNodeLabel <- nodeNodeLabels if nodeNodeLabel.nodeId === id
+        nodeLabel <- nodeNodeLabel.nodeLabel
+      } yield (nodeLabel)
+
+      val relatedObjectNodeLabelQuery = (for {
+        nodeLabel <- nodeLabelQuery
+        relatedObjectNodeLabelEdge <- nodeLabelEdges if relatedObjectNodeLabelEdge.subjectNodeLabelLabel === nodeLabel.label
+        relatedObjectNodeLabel <- relatedObjectNodeLabelEdge.objectNodeLabel
+      } yield (relatedObjectNodeLabel))
+
+      val relatedSubjectNodeLabelQuery = (for {
+        nodeLabel <- nodeLabelQuery
+        relatedSubjectNodeLabelEdge <- nodeLabelEdges if relatedSubjectNodeLabelEdge.objectNodeLabelLabel === nodeLabel.label
+        relatedSubjectNodeLabel <- relatedSubjectNodeLabelEdge.subjectNodeLabel
+      } yield (relatedSubjectNodeLabel))
+
+      val relatedNodeLabelQuery = (relatedObjectNodeLabelQuery ++ relatedSubjectNodeLabelQuery)
+      val relatedNodeLabelWithNodeSourceAction = nodeLabels.withSourceNode(relatedNodeLabelQuery).map {
+        case (nodeLabel, source, nodeLabelNode, nodeLabelNodeSource, nodeLabelNodeLabel) =>
+          (nodeLabel, source.id, nodeLabelNode, nodeLabelNodeSource.id, nodeLabelNodeLabel.label)
+      }.result
+      
+      val relatedNodeLabels = toKgNodeLabels(runSyncTransaction(relatedNodeLabelWithNodeSourceAction)).toList
+
+      // TODO replace inner id order by with pageRank
+      val topEdgesQuery =
```

Comment:
Once this is done please add some documentation as to what this is doing. I'd expect that the explanation of the SQL will take as many lines as the SQL itself.

Reply:
Understood, will do